### ISSUE TITLE
Terraform plan crashes: panic: inconsistent list element types

### ISSUE THREAD

**Issue by Naragorn** [hashicorp/terraform#25625]
panic: inconsistent list element types
[crash.log](https://github.com/hashicorp/terraform/files/4952774/crash.log)
 (cty.Object(map[string]cty.Type{"actions":cty.List(cty.Object(map[string]cty.Type{"category":cty.String, "configuration":cty.Map(cty.String), "input_artifacts":cty.List(cty.String), "name":cty.String, "namespace":cty.String, "output_artifacts":cty.List(cty.String), "owner":cty.String, "provider":cty.String, "region":cty.String, "role_arn":cty.String, "run_order":cty.DynamicPseudoType, "version":cty.String})), "name":cty.String}) then cty.Object(map[string]cty.Type{"actions":cty.List(cty.Object(map[string]cty.Type{"category":cty.String, "configuration":cty.Map(cty.String), "input_artifacts":cty.List(cty.String), "name":cty.String, "namespace":cty.String, "output_artifacts":cty.List(cty.String), "owner":cty.String, "provider":cty.String, "region":cty.String, "role_arn":cty.String, "run_order":cty.Number, "version":cty.String})), "name":cty.String}))

goroutine 710 [running]:
github.com/zclconf/go-cty/cty.ListVal(0xc001d99140, 0x6, 0x6, 0xc0017de180, 0xc0014abaf0, 0x1, 0x1)
        github.com/zclconf/go-cty@v1.2.1/cty/value_init.go:166 +0x438
github.com/zclconf/go-cty/cty/convert.conversionTupleToList.func2(0x56261b973960, 0xc0017c8a20, 0x56261b5005e0, 0xc0017c8a40, 0xc0014abaf0, 0x1, 0x1, 0xc000780000, 0x10, 0x10, ...)
        github.com/zclconf/go-cty@v1.2.1/cty/convert/conversion_collection.go:267 +0x549
github.com/zclconf/go-cty/cty/convert.getConversion.func1(0x56261b973960, 0xc0017c8a20, 0x56261b5005e0, 0xc0017c8a40, 0x0, 0x0, 0x0, 0xc0017df8f0, 0xc0014abac0, 0x56261b973860, ...)
        github.com/zclconf/go-cty@v1.2.1/cty/convert/conversion.go:46 +0x42d
github.com/zclconf/go-cty/cty/convert.retConversion.func1(0x56261b973960, 0xc0017c8a20, 0x56261b5005e0, 0xc0017c8a40, 0xc0014abac0, 0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/zclconf/go-cty@v1.2.1/cty/convert/conversion.go:179 +0x6d
github.com/zclconf/go-cty/cty/convert.Convert(0x56261b973960, 0xc0017c8a20, 0x56261b5005e0, 0xc0017c8a40, 0x56261b973860, 0xc0001c48a0, 0x0, 0x56261b973960, 0xc0017c8a20, 0x56261b5005e0, ...)
        github.com/zclconf/go-cty@v1.2.1/cty/convert/public.go:51 +0x1ac
github.com/hashicorp/terraform/terraform.(*EvalModuleCallArgument).Eval(0xc00141f680, 0x56261b99aaa0, 0xc0018161a0, 0x2, 0x2, 0x58000, 0x51)
        github.com/hashicorp/terraform/terraform/eval_variable.go:77 +0x162
github.com/hashicorp/terraform/terraform.EvalRaw(0x56261b932880, 0xc00141f680, 0x56261b99aaa0, 0xc0018161a0, 0x3c, 0x0, 0x0, 0x3c)
        github.com/hashicorp/terraform/terraform/eval.go:57 +0x133
github.com/hashicorp/terraform/terraform.(*EvalOpFilter).Eval(0xc001471b30, 0x56261b99aaa0, 0xc0018161a0, 0x2, 0x2, 0x0, 0x0)
        github.com/hashicorp/terraform/terraform/eval_filter_operation.go:37 +0x4e
github.com/hashicorp/terraform/terraform.EvalRaw(0x56261b9328c0, 0xc001471b30, 0x56261b99aaa0, 0xc0018161a0, 0xdbb5dcef, 0x100, 0xc001e7a000, 0xc0000b6080)
        github.com/hashicorp/terraform/terraform/eval.go:57 +0x133
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc0011a2660, 0x56261b99aaa0, 0xc0018161a0, 0x2, 0x2, 0x56261a55c687, 0x56261b932f60)
        github.com/hashicorp/terraform/terraform/eval_sequence.go:20 +0xff
github.com/hashicorp/terraform/terraform.EvalRaw(0x56261b932a20, 0xc0011a2660, 0x56261b99aaa0, 0xc0018161a0, 0x56261b5ad560, 0x56261c783b25, 0x56261b525c20, 0xc00185bd20)
        github.com/hashicorp/terraform/terraform/eval.go:57 +0x133
github.com/hashicorp/terraform/terraform.Eval(0x56261b932a20, 0xc0011a2660, 0x56261b99aaa0, 0xc0018161a0, 0xc0011a2660, 0x56261b932a20, 0xc0011a2660, 0x58000)
        github.com/hashicorp/terraform/terraform/eval.go:35 +0x4f
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x56261b76fb80, 0xc00150eb80, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform/terraform/graph.go:90 +0x1344
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc000cd6100, 0x56261b76fb80, 0xc00150eb80, 0xc000e5c540)
        github.com/hashicorp/terraform/dag/walk.go:392 +0x359
created by github.com/hashicorp/terraform/dag.(*Walker).Update
        github.com/hashicorp/terraform/dag/walk.go:314 +0xa90

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform v0.12.28
+ provider.aws v2.70.0

**Comment by alisdair** [comment]
@Naragorn Could you please fill out the full issue template? [Here's a link to the document](https://github.com/hashicorp/terraform/blob/master/.github/ISSUE_TEMPLATE/bug_report.md). In particular, it would be very helpful to see at least the relevant parts of your configuration, so that we can figure out what could be causing this crash. Thanks!

**Comment by adamgoose** [comment]
---
name: Bug report
about: Let us know about an unexpected error, a crash, or an incorrect behavior.

---



### Terraform Version

```
v0.12.28
```

### Terraform Configuration Files
```terraform
variable "apps" {
  description = "Applications to provision, along with their vault config"
  type = list(object({
    name = string
    vault = list(object({
      path         = string
      capabilities = list(string)
    }))
    serviceAccount = object({
      annotations = map(any)
    })
  }))
  default = []
}

// value provided...
  apps = [{
    name = "api"
    vault = [{
      path         = "secret/data/bridge/{env}/{region}/shared/*"
      capabilities = ["read", "list"]
      }, {
      path         = "secret/data/bridge/{env}/{region}/zuul/api"
      capabilities = ["read", "list"]
    }]
    serviceAccount = {
      annotations = {}
    }
    }, {
    name  = "migration"
    vault = []
    serviceAccount = {
      annotations = {
        "eks.amazonaws.com/role-arn" = "arn:aws:iam::OBFUSCATED:role/truss/{truss_env}/{region}/zuul-{env}/zuul-{env}-migration"
      }
    }
  }]
```

### Crash Output

https://gist.github.com/adamgoose/5449e834b4185f3868399bbbb4502f4a

### Expected Behavior

It works.

### Actual Behavior

It crashes.

### Steps to Reproduce

Define a module with the aforementioned variable definition, then use that module providing the aforementioned value.

**Comment by alisdair** [comment]
Thank you for the detailed report! I was able to reproduce the crash on 0.12.28.

This seems to be a duplicate of #21588, which has now been fixed in the upstream library and will be released shortly as part of the 0.13 release. I've verified that the bug is fixed as of the latest 0.13 beta release.
